core/src/autogluon/core/models/ensemble/fold_fitting_strategy.py

```diff
@@ -249,7 +284,21 @@ def _fit(self, model_base, time_start_fold, time_limit_fold, fold_ctx, kwargs):
             X_fold = pd.concat([X_fold, self.X_pseudo], axis=0, ignore_index=True)
             y_fold = pd.concat([y_fold, self.y_pseudo], axis=0, ignore_index=True)
 
-        fold_model.fit(X=X_fold, y=y_fold, X_val=X_val_fold, y_val=y_val_fold, time_limit=time_limit_fold, **kwargs_fold)
+        num_cpus = self.num_cpus
+        num_gpus = self.num_gpus
+        if self.user_resources_per_job is not None:
+            num_cpus = min(self.num_cpus, self.user_resources_per_job.get('num_cpus', math.inf))
+            num_gpus = min(self.num_gpus, self.user_resources_per_job.get('num_gpus', math.inf))
```

Comment:
Why math.inf?

Reply:
Just to make sure that it will use `self.num_cpus` or `self.num_gpus` if user didn't specify resources per job